pkg/pillar/cas/cas.go

```diff
@@ -0,0 +1,66 @@
+package cas
+
+import (
+	"fmt"
+	"io"
+	"time"
+)
+
+//CAS: provides methods to interact with CAS clients
+type CAS interface {
+	//Blob APIs
+	//CheckBlobExists: returns true if the blob exists.
+	CheckBlobExists(blobSha256 string) bool
+	//GetBlobSize: returns size of the blob
+	GetBlobSize(blobSha256 string) (int64, error)
+	//ListBlobs: returns list of blob sha256
+	ListBlobs() ([]string, error)
+	//IngestBlob: ingests a blob from the given reader and assigns the given sha256 to it.
+	IngestBlob(blobSha256 string, reader io.Reader) error
```

Comment:
A few small changes here:

* make the hash argument `blobHash` or similar, i.e. not only sha256
* make it clear that the format for `blobHash` is `<algo>:<hash>`, and only sha256 supported for now
* make it clear that it will validate the blob, and if the hash does not match, it will return an error; if it does, it will ingest it. It doesn't really assign a sha to it; either it works or it doesn't, and we address it by the content.

Reply:
Done.